### PR TITLE
Support new `torch-nos==0.1.0` API with support for default kwargs

### PR DESCRIPTION
**Pull request by spillai**
This PR updates `client.Run()` with the new nos `torch-nos==0.1.0` API
that removes the need for specifying task types. Additionally,
we also introduce an explicit `shm=True` kwarg to `client.Run()` to
enable shared memory. The nos tests have been updated to reflect the
new API as well.

- - -
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/30)

### CODE REVIEW

**Comment by spillai** [comment]
FYI @mkornacker - to be merged once we've released `0.1.0`.

cc @outtanames

File: pixeltable/exec.py [hunk 1]
```diff
@@ -658,8 +658,7 @@ def _exec_cohort(self, cohort: Cohort, rows: DataRowBatch) -> None:
                         _logger.debug(
                             f'Running NOS task {cohort.model_info.task}: '
                             f'batch_size={num_nos_batch_rows} target_res={target_res}')
-                        result = Env.get().nos_client.Run(
-                            task=cohort.model_info.task, model_name=cohort.model_info.name, **kwargs)
+                        result = Env.get().nos_client.Run(cohort.model_info.name, inputs=kwargs, method=cohort.model_info.default_method, shm=True)
```
Comment (mkornacker) [inline pixeltable/exec.py:663]: Do you have docs for ModelSpec? The use of default_method would warrant a comment here.

File: pixeltable/function.py [hunk 2]
```diff
@@ -473,47 +474,61 @@ def register_function(self, module_name: str, fn_name: str, fn: Function) -> Non
...
-        model_info.sort(key=lambda info: info.task.value)
+        models: List[str] = Env.get().nos_client.ListModels()
+        
+        model_info = []
+        for model in models:
+            info = Env.get().nos_client.GetModelInfo(model)
+            # Models have multiple signatures for different tasks.
+            # Here we register all of the model's methods as a separate function.
+            for method in info.signature:
+                minfo = copy.deepcopy(info)
```
Comment (mkornacker) [inline pixeltable/function.py]: uninformative name

File: pixeltable/function.py [hunk 3]
```diff
@@ -473,47 +474,61 @@ def register_function(self, module_name: str, fn_name: str, fn: Function) -> Non
...
         """Register all models supported by the NOS backend as library functions"""
         if self.has_registered_nos_functions:
             return
         self.has_registered_nos_functions = True
-        models = Env.get().nos_client.ListModels()
-        model_info = [Env.get().nos_client.GetModelInfo(model) for model in models]
-        model_info.sort(key=lambda info: info.task.value)
+        models: List[str] = Env.get().nos_client.ListModels()
+        
+        model_info = []
```
Comment (mkornacker) [inline pixeltable/function.py]: missing type hint

File: pixeltable/utils/clip.py [hunk 4]
```diff
@@ -8,11 +8,13 @@
...
-    result = Env.get().nos_client.Run(task=model_info.task, model_name=model_info.name, images=[img.resize((224, 224))])
+    assert model_info.default_method == "encode_image"
+    result = Env.get().nos_client.Run(model_info.name, inputs={"images": [img.resize((224, 224))]}, method=model_info.default_method, shm=True)
     return result['embedding'].squeeze(0)
 
 def embed_text(text: str) -> np.ndarray:
     from pixeltable.functions.text_embedding import openai_clip
-    model_info = FunctionRegistry.get().get_nos_info(openai_clip)
-    result = Env.get().nos_client.Run(task=model_info.task, model_name=model_info.name, texts=[text])
+    model_info, method = FunctionRegistry.get().get_nos_info(openai_clip)
```
Comment (mkornacker) [inline pixeltable/utils/clip.py]: I didn't see a change to get_nos_info().

**Review by mkornacker — CHANGES_REQUESTED** [review]

File: pixeltable/exec.py (hunk 1 above)
Comment (spillai) [reply pixeltable/exec.py:663]: Added comments there with links to docs: https://autonomi-ai.github.io/nos/docs/api/common/spec.html#nos.common.spec.FunctionSignature. Will be updating these shortly.

File: pixeltable/utils/clip.py (hunk 4 above)
Comment (spillai) [reply pixeltable/utils/clip.py]: Good catch, resolved it.

File: pixeltable/function.py (hunk 3 above)
Comment (spillai) [reply pixeltable/function.py]: Addressed.

File: pixeltable/function.py (hunk 2 above)
Comment (spillai) [reply pixeltable/function.py]: Addressed.

**Pull request closed without merging**